src/Twilio/Twilio.cs

```diff
@@ -204,7 +204,7 @@ public static ITwilioRestClient GetNoAuthRestClient()
             }
 
             AuthStrategy noauthstrategy = new NoAuthStrategy();
-           _noAuthRestClient = new TwilioRestClient(_username, _password, accountSid: _accountSid, region: _region, edge: _edge, authstrategy: noauthstrategy)
+           _noAuthRestClient = new TwilioRestClient(_username, _password, authstrategy: noauthstrategy, accountSid: _accountSid, region: _region, edge: _edge)
```

Comment:
Is it the case that order in which the params are passed, is considered? If that's the case, we should add authStrategy as the last param since it won't break any current user who is passing region or edge specifically.

Reply:
Order in which parameters passed is not considered if we mention the name of the parameters. 
There was an issue reported because we added a new optional parameter auth strategy

In this PR, this was implemented as two different overloaded methods. Restored the old method as it was. Added a new method with authstrategy a mandatory parameter. If we add auth strategy as an optional parameter, these two method calls become ambigous and throws error while compiling

So I added auth strategy as mandatory parameter. Mandatory parameters has to be appeared before optional ones

Reply:
Okay that makes sense. Thanks!

Reply:
hey - this change in order absolutlely matters and you have broken your aspnet library with this change

https://github.com/twilio-labs/twilio-aspnet/blob/e13d459b812b6a305144a2677e06e02d36c41083/src/Twilio.AspNet.Core/TwilioClientDependencyInjectionExtensions.cs#L182-L204

this change is not binary compatible with the prior release and there is not MissingMethodExceptions occurring because of this.

Named parameters (and optional parameters) are only syntatic sugar. they are baked in the the caller and the compiler will reorder and subsitute defaults in the calling library.

You need to urgently release an updated version of the aspnet library that is compiled against this version of the package. Or preferably fix the constructor on this package to make it compatible again. 


Also changing a public constructor on a public class is a breaking change, thats a semver breaking change.

Reply:
linking to the issue https://github.com/twilio/twilio-csharp/issues/768